fotoobo/fortinet/fortimanager.py

```diff
@@ -183,14 +194,41 @@ def login(self) -> int:
         """
         Login to the FortiManager.
 
-        If the login to the FortiManager was successful it stores the session key in rhe object
+        If the login to the FortiManager was successful it stores the session key in the object.
         We do not use requests.session as the session key is just a string which is saved directly.
 
         Returns:
             int: status code from the FortiManager login
         """
         status: int = 401
-        if self.username and self.password:
+
+        if self.session_path:
+            session_file = Path(self.session_path).expanduser() / f"{self.hostname}.key"
+            try:
+                log.debug("loading session key from file %s", session_file)
+                with session_file.open(encoding="UTF-8") as file:
+                    self.session_key = file.read()
+                    payload = {
+                        "method": "get",
+                        "params": [{"url": "/sys/status"}],
+                        "session": self.session_key,
+                    }
+                    response = super().api("post", payload=payload)
+                    status = response.status_code
+                    if (
+                        response.status_code == 200
+                        and response.json()["result"][0]["status"]["code"] == 0
+                    ):
+                        log.debug("Session key is still valid")
+                    else:
+                        log.debug("Session key is invalid")
+                        self.session_key = ""
+
+            except FileNotFoundError:
```

Comment:
I think we should use if here, because this will happen at least for the first time fotoobo is run with session_key storage, so not entirely "unexpected" or partly random.

Reply:
Of course I fixed that. I'm with you that using if here is a better idea.